### PR TITLE
Catch exceptions when writing XML doc files in VB

### PR DESCRIPTION
**Pull request by agocke**
This is the same fix as was implemented for C# in #13806, but for VB.

/cc @AlekseyTs @dotnet/roslyn-compiler

### CODE REVIEW

**Comment by agocke** [comment]
test windows_eta_open_prtest please

File: src/Compilers/VisualBasic/Portable/Compilation/DocumentationComments/DocumentationCommentCompiler.vb [hunk 1]
```diff
@@ -68,14 +68,19 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
...
                             diagnostics, filterTree, filterSpanWithinTree, preferredCulture:=Nothing, cancellationToken:=cancellationToken)
 
-                    compiler.Visit(compilation.SourceAssembly.GlobalNamespace)
-                    Debug.Assert(compiler._writer.IndentDepth = 0)
-                End Using
+                        compiler.Visit(compilation.SourceAssembly.GlobalNamespace)
+                        Debug.Assert(compiler._writer.IndentDepth = 0)
+                        writer?.Flush()
+                    End Using
+                Catch ex As Exception
```
Comment (jasonmalinowski) [inline src/Compilers/VisualBasic/Portable/Compilation/DocumentationComments/DocumentationCommentCompiler.vb]: Should this catch something a bit more...precise?

Comment (jaredpar) [inline src/Compilers/VisualBasic/Portable/Compilation/DocumentationComments/DocumentationCommentCompiler.vb]: No.  Pokemon exception handling.

**Comment by jasonmalinowski** [comment]
@dotnet-bot test windows_release_unit32_prtest please and test windows_release_unit64_prtest. (These are new jobs that are now also required.)

**Comment by jasonmalinowski** [comment]
@dotnet-bot test windows_release_unit32_prtest please and test windows_release_unit64_prtest please. (These are new jobs that are now also required.)

File: src/Compilers/VisualBasic/Portable/Compilation/DocumentationComments/DocumentationCommentCompiler.vb (hunk 1 above)
Comment (agocke) [inline src/Compilers/VisualBasic/Portable/Compilation/DocumentationComments/DocumentationCommentCompiler.vb]: Nope. Pokemon. Only way to go.

File: src/Compilers/VisualBasic/Portable/Errors/Errors.vb [hunk 2]
```diff
@@ -1686,6 +1686,7 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
         ERR_PeWritingFailure = 37256
 
         ERR_OptionMustBeAbsolutePath = 37257
+        ERR_DocFileGen = 37258
 
         ' available 37258
```
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Portable/Errors/Errors.vb]: >37258 [](start = 20, length = 5)

Not available after the change.

**Comment by AlekseyTs** [comment]
LGTM

File: src/Compilers/VisualBasic/Portable/Compilation/DocumentationComments/DocumentationCommentCompiler.vb [hunk 3]
```diff
@@ -68,14 +68,19 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                     writer = New StreamWriter(xmlDocStream, New UTF8Encoding(True, False), bufferSize:=&H400, leaveOpen:=True)
                 End If
 
-                Using writer
-                    ' TODO: get preferred culture from compilation(?)
-                    Dim compiler As New DocumentationCommentCompiler(If(assemblyName, compilation.SourceAssembly.Name), compilation, writer, True, False,
+                Try
+                    Using writer
+                        ' TODO: get preferred culture from compilation(?)
```
Comment (AdamSpeight2008) [inline src/Compilers/VisualBasic/Portable/Compilation/DocumentationComments/DocumentationCommentCompiler.vb]: Is this comment still relevant? If not, then remove.

**Comment by AdamSpeight2008** [comment]
If the condition on [line 67](https://github.com/agocke/roslyn/blob/25b5f9c1bd417cf556561dbb4f63351699c44b4f/src/Compilers/VisualBasic/Portable/Compilation/DocumentationComments/DocumentationCommentCompiler.vb#L67) is false what happens? Couldn't we exit the sub early?

File: src/Compilers/VisualBasic/Portable/Errors/Errors.vb [hunk 4]
```diff
@@ -1686,8 +1686,8 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
         ERR_PeWritingFailure = 37256
 
         ERR_OptionMustBeAbsolutePath = 37257
+        ERR_DocFileGen = 37258
```
Comment (AdamSpeight2008) [inline src/Compilers/VisualBasic/Portable/Errors/Errors.vb]: Clashes with       `ERR_TupleExplicitNamesOnAllMembersOrNone = 37258`

File: src/Compilers/VisualBasic/Portable/Compilation/DocumentationComments/DocumentationCommentCompiler.vb (hunk 3 above)
Comment (agocke) [inline src/Compilers/VisualBasic/Portable/Compilation/DocumentationComments/DocumentationCommentCompiler.vb]: Still relevant -- we're still using None as the preferred culture.

**Comment by agocke** [comment]
@AdamSpeight2008 Just because we don't write the stream doesn't mean the compiler couldn't generate diagnostics that would be added to the bag.

File: src/Compilers/VisualBasic/Portable/Errors/Errors.vb (hunk 4 above)
Comment (AdamSpeight2008) [inline src/Compilers/VisualBasic/Portable/Errors/Errors.vb]: #13799

**Comment by AdamSpeight2008** [comment]
You can move the If condition onto the `Using Block`
```vb.net
            Friend Shared Sub WriteDocumentationCommentXml(compilation As VisualBasicCompilation,
                                                           assemblyName As String,
                                                           xmlDocStream As Stream,
                                                           diagnostics As DiagnosticBag,
                                                           cancellationToken As CancellationToken,
                                                           Optional filterTree As SyntaxTree = Nothing,
                                                           Optional filterSpanWithinTree As TextSpan? = Nothing)
                Try
                    Using writer = If(xmlDocStream IsNot Nothing AndAlso xmlDocStream.CanWrite,
                                        New StreamWriter(xmlDocStream, New UTF8Encoding(True, False), bufferSize:=&H400, leaveOpen:=True),
                                        Nothing)

                        ' TODO: get preferred culture from compilation(?)
                        Dim compiler As New DocumentationCommentCompiler(If(assemblyName, compilation.SourceAssembly.Name),
                                                                          compilation, writer, True, False, diagnostics,
                                                                          filterTree, filterSpanWithinTree,
                                                                          preferredCulture:=Nothing,
                                                                          cancellationToken:=cancellationToken
                                                                        )

                        compiler.Visit(compilation.SourceAssembly.GlobalNamespace)
                        Debug.Assert(compiler._writer.IndentDepth = 0)
                    End Using
                Catch ex As Exception
                    diagnostics.Add(ERRID.ERR_DocFileGen, Location.None, ex.Message)
                End Try
                For Each tree In compilation.SyntaxTrees
                    MislocatedDocumentationCommentFinder.ReportUnprocessed(tree, filterSpanWithinTree, diagnostics, cancellationToken)
                Next
            End Sub
```

**Pull request merged**